### DISCUSSION TITLE
[Question] Recommended way to send traces to Apollo Studio or Datadog?

### DISCUSSION THREAD

**Discussion by jan-opal** [99designs/gqlgen#1787] [Q&A]
Not really and issue, but hoping for some guidance. 

I can see traces in GraphiQL perfectly, but would love to aggregate them in an external tool like Apollo Studio or Datadog. 

Is there any plans to add support for that out of the box? Any guidance on how to achieve that currently?

Thanks all :)

**Comment by tprebs** [comment]
Apollo tracing is designed to be added to the query response and not aggregated to Apollo Studio or Datadog.

> Apollo Tracing works by including data in the extensions field of the GraphQL response, which is reserved by the GraphQL spec for extra information that a server wants to return. That way, you have access to performance traces alongside the data returned by your query.

I am currently looking at creating an extension to gqlgen (similar to Apollo tracing) to send Metrics to Apollo Studio. I believe Datadog also supports consuming the format for metrics for Apollo Studio so implementing an extension for Apollo Studio should cover Apollo Studio and Datadog.

**Comment by tprebs** [comment]
If you want to play around with aggregating the Apollo tracing messages, the implementation for gqlgen Apollo Tracing is here https://github.com/99designs/gqlgen/blob/master/graphql/handler/apollotracing/tracer.go

**Reply by dgocoder** [reply]
Curious if you have made any progress on this?

**Reply by tprebs** [reply]
I made quite a bit of progress on this a couple of months ago but it has been pushed into the background by a few other priorities. I will get back on it over the next week or so

**Reply by mlaflamm** [reply]
It look like there was so progress recently on this #2132

**Reply by AaronBuxbaum** [reply]
Has there been any progress made in this direction since? We're trying to come up with a decent way to get Apollo Studio; my current thinking is to wrap our schema in federation and run everything through [Router](https://www.apollographql.com/docs/router/) but this is quite a substantial change and a decent hit to performance vs what is likely much cleaner as a plugin directly into the server itself. Any advice?
